anoncreds/tests/utils/mock.rs

```diff
@@ -0,0 +1,443 @@
+use super::anoncreds::{IssuerWallet, Ledger, ProverWallet, StoredCredDef, StoredRevDef};
+use std::{
+    collections::{HashMap, HashSet},
+    fs::create_dir,
+};
+
+use anoncreds::{
+    data_types::anoncreds::{
+        cred_def::{CredentialDefinition, CredentialDefinitionId},
+        cred_offer::CredentialOffer,
+        credential::Credential,
+        presentation::Presentation,
+        rev_reg::{NonRevProofWarnings, RevocationRegistry, RevocationRegistryId},
+        rev_reg_def::RevocationRegistryDefinitionId,
+        schema::{Schema, SchemaId},
+    },
+    issuer, prover,
+    tails::{TailsFileReader, TailsFileWriter},
+    types::{
+        CredentialDefinitionConfig, CredentialRequest, CredentialRevocationConfig,
+        CredentialRevocationState, MakeCredentialValues, PresentCredentials, PresentationRequest,
+        RegistryType, RevocationStatusList, SignatureType,
+    },
+    verifier,
+};
+use bitvec::bitvec;
+
+// {cred_def_id: {
+//       schema_id, credential_values, support_revocation, rev_reg_id, rev_idx
+// }}
+pub type IsserValues<'a> =
+    HashMap<&'a str, (&'a str, HashMap<&'a str, &'a str>, bool, &'a str, u32)>;
+
+// {cred_def_id: {
+//       attribute_per_credential, predicate_for_credential }}
+pub type ProverValues<'a> = HashMap<&'a str, (Vec<&'a str>, Vec<&'a str>)>;
+
+#[derive(Debug)]
+pub struct Mock<'a> {
+    pub issuer_wallets: HashMap<&'a str, IssuerWallet<'a>>,
+    pub prover_wallets: HashMap<&'a str, ProverWallet<'a>>,
+    pub ledger: Ledger<'a>,
+    pub tails_path: &'a str,
+    pub max_cred_num: u32,
+}
+
+impl<'a> Mock<'a> {
```

Comment:
I have never really done any intensive mocking with Rust, but would it not be easier to pull in a dependency for this? I have no issue with keeping this, but we might have a bit less code with a mock dependency.